### PR TITLE
feat(Lungo-UI): Centralized error handling

### LINKED ISSUE #648: [UI]: Implement proper UI error handling
https://github.com/agntcy/coffeeAgntcy/pull/614#discussion_r3427297581

Let's implement a proper user facing UI error handling so we don't fail silently.

| Area | Current state |
|------|----------------|
| **React ErrorBoundary** | **None** — no `ErrorBoundary`, `componentDidCatch`, or `react-error-boundary` usage |
| **Request error parsing** | Partial — `parseApiError` / `parseFetchError` in `utils/const.ts`, but duplicated `messageFromAxiosResponse` in Identity/Directory APIs |
| **User-visible errors** | Inconsistent — some flows show inline errors (modals, feeds), others log-only or generic copy |
| **Developer-visible errors** | **`logger` is dev-only** — all `logger.apiError` / `logger.error` calls are **no-ops in production**; `unsafeLogger` exists but is **never used** |
| **Global UI error channel** | **None** — no toast/snackbar/banner store; each component rolls its own `useState(error)` |

Tasks:

***Foundation (1–4)***
* Shared request error types and parsers — Add LungoRequestError, consolidate parseFetchError / parseAxiosError, unit tests, and temporary re-exports from const.ts.

* Request error reporting — Add reportRequestError that uses logger in dev and redacted unsafeLogger in production.

* Global error notification UI — Zustand store (pushError, dismiss, clearAll), accessible banner component, reportUiError, mounted in AppProviders.

* React ErrorBoundary — Fallback UI with retry, logging via reportUiError, wrap app shell (optional nested boundary around the graph).

***API migrations (5–6)***
* Identity & Directory APIs — Remove duplicated messageFromAxiosResponse; use shared parsers and LungoRequestError.

* Agentic workflows client & catalog — Centralize fetch/axios errors; surface bootstrap failures and real catalog error text in the sidebar.

***User-visible bug fixes (7–10)***
* Agentic graph & SSE failures — Show graph-level errors when bootstrap, SSE reconnect (6 attempts), or topology refetch exhaust retries; wire reportRequestError.

* Auction streaming parity — Match group/recruiter: error useEffect, clear loading on early errors, try/catch around connect.

* Chat apiError unification — Set apiError for all chat patterns; show server detail in ChatAreaMessageThread instead of generic copy.

* Transport labels & suggested prompts — Stop silent failures; report errors and optionally show non-blocking banners; cap prompt retries.

***Cleanup & tests (11–12)***
* Streaming stores migration — Replace ad hoc logger.apiError in auction/group/recruiter stores, useApp, useAgentAPI with shared reporting.

* Integration tests — Parser edge cases (FastAPI validation, HTML strip), ErrorBoundary fallback, optional notification store tests.

### PR DESCRIPTION
**Pull request by misi-bp**
Overhaul Lungo frontend networking and failure handling: replace **axios** with a shared **fetch** stack, centralize URLs and request targets, report API errors consistently, and surface actionable messages across chat, streaming, agentic graph, and modals. Follow-up commits on this branch align **`reportRequestError`** labels with fetch targets, fix graph OASF/identity edge cases, and document the pipeline for maintainers.

---

### HTTP client and URL layer

- Replace **axios** with **`httpFetch`** and helpers: **`fetchJson`**, **`fetchNdjsonStream`**, **`fetchSse`** (shared SSE frame parsing).
- Introduce **`HttpError`**, **`parseHttpError`**, and **`endpointLabel`** on errors for logging and UI metadata.
- Centralize routes in **`urls.ts`** (`apiPaths`, env-backed bases) and **`httpRequestTargets.ts`** / **`HttpRequestTarget`** builders; migrate **`agenticWorkflowsClient`** and catalog consumers to fetch.
- **`workflowChatRouting`** resolves prompt/stream/suggested-prompt URLs from **`WorkflowSummary.chat_api_target`** and streaming flags.

### Error reporting and global UI

- **`reportRequestError`**: structured dev/prod logging; optional **`userMessage`** → global notification banner via **`errorNotificationStore`**.
- **`reportUiError`**, **`ErrorBoundary`** with recoverable fallback, and error notification banner on **`RootPage`**.
- Logs and banner **`source`** prefer **`HttpError.endpointLabel`** when the transport layer set it.

### Chat and streaming

- Agent prompt (JSON and NDJSON), auction / group / recruiter streaming stores unified on **`fetchNdjsonStream`** + **`reportRequestError`**; shared **`ndjsonStreamingStatus`** / stream teardown hardening.
- Pattern reference documentation and pattern chat use agentic-workflows request builders; errors reported at hook boundaries (no duplicate logging in stream UI callbacks).
- Suggested prompts, transport config, and catalog/bootstrap/SSE failures surfaced with user-visible messages instead of silent failures.
- **`useApp`** split into focused hooks (`useAppPromptHandlers`, streaming effects, workflow catalog, pattern reference, etc.).

### Graph, modals, and identity

- Unified modal error presentation (**`LoadingErrorState`**, compact negative empty states) for badge, policy, OASF, and about flows.
- **`reportRequestError`** in graph modals uses the same **`endpointLabel`** as the fetch (**`badgeDetailsRequest`**, **`oasfRecordRequest`**, pattern chat/documentation builders).
- **`useNodeTransportInterfaces`**: pass workflow **`chat_api_target`** into **`fetchOasfRecord`** (fixes logistics OASF 404s to exchange port); cache empty transport results to avoid refetch loops.
- **Directory node**: dedicated OASF modal content (directory URL/version); no identity-apps badge/policy actions; safe endpoint labels in identity modal **`catch`** when slug resolution fails.
- Directory information modal field typography aligned with **Info** modal (**`body2`** label/value rows).

### UI polish and dependencies

- Conditional **TransportRail** padding on **`CustomNode`** only; dark-mode resting rail label contrast; compact-group layout padding split.
- **`Content-Type: application/json`** on JSON POST bodies (fixes 422 on **`/agent/prompt`** after axios removal).
- Bump **`@open-ui-kit/core`** to **2.2.3**; prefer OUK imports over direct MUI where touched; security-related dependency updates and **tsconfig** adjustments.

### Documentation

- Add **[http-and-error-handling.md](./http-and-error-handling.md)** — URL builders, transport helpers, **`reportRequestError`** contract, and per-feature error presentation map.

### Tests

- Coverage for **`parseHttpError`**, **`reportRequestError`**, **`ErrorBoundary`**, workflow chat routing, agentic workflows API parsing, topology/identity UI map, and related regressions.

---

## Issue link

Fixes #648

## Type of change

- [x] Bugfix
- [x] New feature
- [ ] Breaking change
- [x] Refactor
- [x] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](/agntcy/coffeeAgntcy/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented ([http-and-error-handling.md](./http-and-error-handling.md))
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: coffeeAGNTCY/coffee_agents/lungo/frontend/src/api/http/fetchNdjsonStream.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      const { done, value } = await reader.read()
+      if (done) break
+
+      buffer += decoder.decode(value, { stream: true })
+
+      if (splitMode === "lines") {
+        buffer = consumeNdjsonLines(
+          buffer,
+          (parsed, raw) => {
+            if (onLine(parsed, raw) === "stop") {
```
Comment (delthazor) [inline coffeeAGNTCY/coffee_agents/lungo/frontend/src/api/http/fetchNdjsonStream.ts:94]: On early stops like this, consider closing the reader stream manually with reader.cancel().
GC will probably close it anyways, but why not get back the resources sooner, if we can...

File: coffeeAGNTCY/coffee_agents/lungo/frontend/src/errors/ui/errorNotificationStore.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        message: input.message,
+        source: input.source,
+      }
+
+      set((state) => ({
+        notifications: [...state.notifications, notification],
+      }))
+
+      return id
+    },
```
Comment (delthazor) [inline coffeeAGNTCY/coffee_agents/lungo/frontend/src/errors/ui/errorNotificationStore.ts]: If I understand correctly this can add indefinitely. There is no duplication checks and count or size limit. Consider adding some flood protection, so when an error happens multiple time in quick succession, the banner does not overload.

**Review by delthazor — COMMENTED** [review]
The solution is comprehensive, and very solid. I think it is good overall, I just had small nitpicks.

**Review by pregnor — DISMISSED** [review]
LGTM at a glance, looked good on the live show

**Comment by misi-bp** [comment]
image

image

image

image

File: coffeeAGNTCY/coffee_agents/lungo/frontend/src/errors/ui/errorNotificationStore.ts (hunk 2 above)
Comment (misi-bp) [reply coffeeAGNTCY/coffee_agents/lungo/frontend/src/errors/ui/errorNotificationStore.ts]: nice one, thanks!

File: coffeeAGNTCY/coffee_agents/lungo/frontend/src/api/http/fetchNdjsonStream.ts (hunk 1 above)
Comment (misi-bp) [reply coffeeAGNTCY/coffee_agents/lungo/frontend/src/api/http/fetchNdjsonStream.ts:94]: done

File: coffeeAGNTCY/coffee_agents/lungo/frontend/src/errors/ui/errorNotificationStore.ts (hunk 2 above)
Comment (misi-bp) [reply coffeeAGNTCY/coffee_agents/lungo/frontend/src/errors/ui/errorNotificationStore.ts]: done

**Review by pregnor — DISMISSED** [review]
LGTM still, just fix the CI issues

**Review by pregnor — DISMISSED** [review]

**Review by pregnor — APPROVED** [review]

**Review by delthazor — APPROVED** [review]
Pretty robust, let's see it in action
